src/Console/CommandRunner.php

```diff
@@ -200,6 +240,11 @@ protected function getShell(ConsoleIo $io, CommandCollection $commands, $name)
      */
     protected function createShell($className, ConsoleIo $io)
     {
-        return new $className($io);
+        if (is_subclass_of($className, Shell::class)) {
+            return new $className($io);
+        }
+
+        // Command class
+        return new $className();
```

Comment:
Is the doc block `@return \Cake\Console\Shell` then correct?

Reply:
Fixed now.